### PR TITLE
Use `curly_braces_in_flow_control_structures` for `material`

### PR DESCRIPTION
**Pull request by guidezpl**
This doesn't turn on the lint yet.

#104498

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM.

I am in favor of turning this on for the whole repository. :)

**Comment by goderbauer** [comment]
> It looks like this pull request may not have tests.

Did you also run this over `tests/material`? Surprised that there wouldn't be a single test violating this.

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -5151,11 +5151,11 @@ void main() {
     String clipboardContent = '';
     tester.binding.defaultBinaryMessenger
       .setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
-        if (methodCall.method == 'Clipboard.setData')
-          // ignore: avoid_dynamic_calls
```
Comment (werainkhatri) [inline packages/flutter/test/material/text_field_test.dart:5155]: here

File: packages/flutter/test/material/text_field_test.dart [hunk 2]
```diff
@@ -5276,11 +5277,11 @@ void main() {
...
```
Comment (werainkhatri) [inline packages/flutter/test/material/text_field_test.dart:5280]: here

**Review by werainkhatri — COMMENTED** [review]
maybe `ignore: avoid_dynamic_calls` was removed unintentionally?

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 3]
```diff
@@ -1928,10 +1972,12 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
   }
 
   Color _getFillColor(ThemeData themeData) {
-    if (decoration!.filled != true) // filled == null same as filled == false
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/input_decorator.dart:1931]: fix me

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 4]
```diff
@@ -419,8 +423,9 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
       canvas.drawArc(Offset.zero & size, 0, _sweep, false, backgroundPaint);
     }
 
-    if (value == null) // Indeterminate
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/progress_indicator.dart:422]: fix me

**Comment by guidezpl** [comment]
Indeed, comments were removed in 4 places, I'll add them back. Something to keep in mind for future PRs.

**Comment by werainkhatri** [comment]
> Indeed, comments were removed in 4 places, I'll add them back. Something to keep in mind for future PRs.

was this because of the use of `dart fix` or a code action?

**Comment by guidezpl** [comment]
This was because of `dart fix`.

**Pull request merged**